ocs_ci/helpers/dr_helpers.py

```diff
@@ -1322,3 +1328,168 @@ def add_label_to_appsub(workloads, label="test", value="test1"):
+        name = placement["metadata"]["name"]
+        if (name != "all-openshift-clusters") and (name != "global"):
+            namespace = placement["metadata"]["namespace"]
+            params = f"""[{{"op": "remove", "path": "{constants.EXPERIMENTAL_ANNOTATION_PATH}"}}]"""
+            cmd = f"oc patch {constants.PLACEMENT} {name} -n {namespace} -p '{params}' --type=json"
+            run_cmd(cmd)
+
+    config.switch_ctx(old_ctx)
+
+
+def apply_drpolicy_to_workload(workload, drcluster_name):
+    """
+    Function for applying drpolicy to indiviusual workload
+
+    Args:
+        workload(List): List of workload objects
+        drcluster_name(str): Name of the DRcluster on which workloads belongs
+
+    """
+    for wl in workload:
+        drpc_yaml_data = templating.load_yaml(wl.drcp_data_yaml.name)
+        logger.info(drpc_yaml_data)
+        if wl.workload_type == constants.SUBSCRIPTION:
+            drpc_yaml_data["metadata"]["namespace"] = wl.workload_namespace
+        drpc_yaml_data["spec"]["preferredCluster"] = drcluster_name
+        templating.dump_data_to_temp_yaml(drpc_yaml_data, wl.drcp_data_yaml.name)
+        config.switch_acm_ctx()
+        wl.add_annotation_to_placement()
+        run_cmd(f"oc create -f {wl.drcp_data_yaml.name}")
+
+
+def replace_cluster(workload, primary_cluster_name, secondary_cluster_name):
+
+    """
+    Function to do core replace cluster task
+
+    Args:
+        workload(List): List of workload objects
+        primary_cluster_name (str): Name of the primary DRcluster
+        secondary_cluster_name(str): Name of the secondary DRcluster
+
+    """
+
+    # Delete dr cluster
+    config.switch_acm_ctx()
+    run_cmd(cmd=f"oc delete drcluster {primary_cluster_name} --wait=false")
+
+    # Disable DR on hub for each app
+    disable_dr_from_app(secondary_cluster_name)
+    logger.info("DR configuration is successfully disabled on each app")
+
+    # Remove DR configuration from hub and surviving cluster
+    logger.info("Running Remove DR configuration script..")
+    run_cmd(cmd=f"chmod +x {constants.REMOVE_DR_EACH_MANAGED_CLUSTER}")
+    run_cmd(cmd=f"sh {constants.REMOVE_DR_EACH_MANAGED_CLUSTER}")
+
+    sample = TimeoutSampler(
+        timeout=300,
+        sleep=5,
+        func=run_cmd_verify_cli_output,
+        cmd="oc get namespace openshift-operators",
+        expected_output_lst={"openshift-operators", "Active"},
+    )
+    if not sample.wait_for_func_status(result=True):
+        raise Exception("Namespace openshift-operators is not created")
+
+    # add label to openshift-opeartors namespace
+    ocp_obj = ocp.OCP(kind="Namespace")
+    label = "openshift.io/cluster-monitoring='true'"
+    ocp_obj.add_label(resource_name=constants.OPENSHIFT_OPERATORS, label=label)
+
+    # Detach old primary
+    run_cmd(cmd=f"oc delete managedcluster {primary_cluster_name}")
+
+    # Verify old primary cluster is dettached
+    expected_output = primary_cluster_name
```

Comment:
why do we need this extra variable? can't we directly access primary_cluster_name in line #1451